source/CIApp.cpp

```diff
@@ -191,14 +175,11 @@ void CoreImpactApp::update(float timestep) {
  */
 void CoreImpactApp::onSuspend() {
     // save settings file
-    std::shared_ptr<cugl::JsonWriter> _writer = JsonWriter::alloc(Application::getSaveDirectory().append("settings.json"));
-    std::shared_ptr<cugl::JsonValue> _settings = JsonValue::allocObject();
-    _settings->appendValue("PlayerName", _menu.getPlayerName());
-    _settings->appendValue("Volume", _menu.getVolume());
-    _settings->appendValue("MusicOn", _menu.isMusicOn());
-    _settings->appendValue("ParallaxOn", _menu.isParallaxOn());
+    auto _writer = JsonWriter::alloc(Application::getSaveDirectory().append("playersettings.json"));
```

Comment:
Is there a reason you're writing the player settings in onSuspend and onShutdown?

Reply:
The player settings was resetting on every new game instance. Saving the settings on both occasions allows the subsequent game instances to load in the saved settings instead of resetting.

Reply:
okay I think you can get of the onShutdown saving then? I think?

Reply:
So on the iphone, the program does not go through onShutdown due to how the phone handles app shutdown/suspend. The UI forces app to suspend before shutting down which prevents the program from triggering the onShutdown method.